d3b_dff_cli/modules/validation/check_manifest.py

```diff
@@ -1,110 +1,127 @@
+        elif isinstance(v, list):
+            schema[k] = [item.lower() if isinstance(item, str) else item for item in v]
+        elif isinstance(v, str):
+            schema[k] = v.lower()
+    return schema
 
-    # Iterate through each row in the DataFrame and perform validation
-    validation_failed = False
-    for index, row in manifest_data.iterrows():
-        is_valid, messages = validate_row(row, validation_rules)
+def validate_data(df, schema_json):
+    """
+    Validate the DataFrame against the schema.
+    """
+    valid = True
+    errors = []
+    custom_rules = schema_json.get('custom_rules', {})
+    
+    for index, row in df.iterrows():
+        experiment_strategy = row.get("experiment_strategy", "").lower()
+        platform = row.get("platform", "").lower()
+        
+        if platform == "pacbio":
+            rule_type = "pacbio_longread_rules"
+        else:
+            if experiment_strategy in ["wgs", "wxs", "wes", "target sequencing", "panel", "target"]:
+                rule_type = "DNAseq_rules"
+            elif experiment_strategy in ["rna-seq", "rnaseq", "mirna-seq", "mirnaseq"]:
+                rule_type = "RNAseq_rules"
+            elif experiment_strategy in ["scrna-seq", "snran-seq", "scrnaseq", "snranseq"]:
+                rule_type = "single_cell_rules"
+            elif experiment_strategy in ["methtlation", "methylation microarray"]:
+                rule_type = "methylation_rules"
+            else:
+                raise ValueError(f"Unsupported experiment_strategy for Row {index + 1}")
+        
+        schema = schema_json.get(rule_type, {})
+        
+        # Filter out fields not in the schema fields for combined manifest
+        row_dict = row.to_dict()
+        filtered_row_dict = {k: v for k, v in row_dict.items() if k in schema}
+        
+        v = CustomValidator(schema, custom_rules)
+        (is_valid, out_error) = v.validate(filtered_row_dict)
+        
         if not is_valid:
-            error_message = "Validation Failed For Row {0}:\n{1}".format(index + 1, '\n'.join(messages))
-            print(error_message,"\n")
-            validation_failed = True
-    if not validation_failed:
-        print("Validation Passed: All rows are valid.")
+            valid = False
+            errors.append({
+                'row': index + 1,
+                'errors': out_error
+            })
+    
+    return valid, errors
+
+def main(args):
+    """
+    Main function to load schema, validate data, and print the validation report.
+    """
+    with open(validation_schema, 'r') as f:
+        schema = json.load(f)
+    
+    schema_json = convert_schema_to_lowercase(schema)
+
+    # Load and preprocess the data
+    df = load_data(args.manifest_file)
+
+    # Validate the data
+    valid, errors = validate_data(df, schema_json)
+
+    # Print validation report
+    if valid:
+        print("====Validation Passed====\n  All rows are valid.")
+    else:
```

Comment:
I don't know if this is what we decided. I thought we decided to leave all validations as pass or fail, then check with the people sending us the data, then once the failures are resolved either redo the validation or skip validation.

Reply:
I’m open to that. And yes, even if we define the file size validation as a warning message, it’s still a kind of failure since it needs to be checked.
I can recovered it into two validation summaries: passed or failed.

Reply:
As discussed with Alex on Slack, I will keep this unchanged.